server/services/phase.js

```diff
@@ -7,93 +7,95 @@ const { getSiteByID } = require('./employers');
-    sitePhase.end_date = dayjs.utc(sitePhase.end_date).format('YYYY/MM/DD');
+  /**
+   * @typedef {Object} sitePhasesResponse  Internal type for DB response to the `getAllSitePhases` query
+   * @property {number} id                 Internal ID of the phase
+   * @property {string} name               Name of the phase
+   * @property {Date} start_date           Start date of the phase
+   * @property {Date} end_date             End date of the phase
+   * @property {number} allocation         Number of employees allocated for this phase at this site
+   * @property {string} hcap_hires         Number of HCAP employees hired (as a string)
+   * @property {string} non_hcap_hires     Number of non-HCAP employees hired (as a string)
+   */
 
-    // remove join attributes
-    delete sitePhase.site_phase_allocation;
-    delete sitePhase.archivedJoin;
-    delete sitePhase.hires;
+  /**
+   * Raw result from a custom query.
+   *
+   * This query essentially performs the following actions:
+   * * Grabs all phases
+   * * Finds the phase allocations for each phase at this site
+   * * For each phase, counts the number of currently hired employees within its timespan
+   * * Splits these hires into HCAP and non-HCAP hires
+   * @type {sitePhasesResponse[]}
+   */
+  const sitePhases = await dbClient.runRawQuery(
+    `
+    SELECT 
+      phase.id, 
+      phase.name, 
+      phase.start_date, 
+      phase.end_date, 
+      spa.allocation, 
+      count(ps.id) FILTER (
+        WHERE 
+          ps.data ->> 'nonHcapOpportunity' = 'false'
+      ) AS hcap_hires, 
+      count(ps.id) FILTER (
+        WHERE 
+          ps.data ->> 'nonHcapOpportunity' = 'true'
+      ) AS non_hcap_hires 
+    FROM 
+      phase 
+      LEFT JOIN site_phase_allocation spa ON phase.id = spa.phase_id AND spa.site_id = '$1'
+      LEFT JOIN participants_status ps ON
+        ps.DATA ->> 'site' = '$1' AND ps."current"
+        AND ps."current"
+        AND to_date(
+          ps.data ->> 'hiredDate', 'YYYY/MM/DD'
+        ) BETWEEN COALESCE(
+          spa.start_date, phase.start_date
+        ) 
+      AND COALESCE(spa.end_date, phase.end_date) 
+    GROUP BY 
+      phase.id, 
+      spa.id
+    `,
+    [site.siteId]
+  );
 
-    return {
-      id: sitePhase.id,
-      phaseName: sitePhase.name,
-      startDate: sitePhase.start_date,
-      endDate: sitePhase.end_date,
-      allocation: sitePhase.allocation,
-      remainingHires: (sitePhase.allocation ?? 0) - sitePhase.hcapHires,
-      hcapHires: sitePhase.hcapHires,
-      nonHcapHires: sitePhase.nonHcapHires,
-    };
-  });
-  return phaseData;
+  // Transform data format and return it
+  return sitePhases.map((phase) => ({
+    id: phase.id,
+    phaseName: phase.name,
+    startDate: dayjs.utc(phase.start_date).format('YYYY/MM/DD'),
+    endDate: dayjs.utc(phase.end_date).format('YYYY/MM/DD'), // strips time/timezone from date and formats it
+    allocation: phase.allocation,
```

Comment:
Isn't the data already being stored without time/timzone for start and end date?

Reply:
ah okay! Thanks for the clarification